lib/domain.js

```diff
@@ -313,13 +323,23 @@ Domain.prototype.add = function(ee) {
     }
   }
 
-  ee.domain = this;
+  Object.defineProperty(ee, 'domain', {
+    configurable: true,
+    enumerable: false,
+    value: this,
+    writable: true
+  });
   this.members.push(ee);
 };
 
 
 Domain.prototype.remove = function(ee) {
-  ee.domain = null;
+  Object.defineProperty(ee, 'domain', {
+    configurable: true,
+    enumerable: false,
+    value: null,
+    writable: true
+  });
```

Comment:
This call should not be required. If the resource once had a domain set, it should already be non-enumerable.